api/v1alpha1/tls_types.go

```diff
@@ -131,3 +135,55 @@ type ClientValidationContext struct {
 	// +optional
 	CACertificateRefs []gwapiv1.SecretObjectReference `json:"caCertificateRefs,omitempty"`
 }
+
+// Configure Envoy proxy how to handle the x-forwarded-client-cert (XFCC) HTTP header.
+type ForwardClientCert struct {
+	// Envoy Proxy mode how to handle the x-forwarded-client-cert (XFCC) HTTP header.
+	// +optional
+	Mode *ForwardMode `json:"mode,omitempty"`
+
+	// Specifies the fields in the client certificate to be forwarded on the x-forwarded-client-cert (XFCC) HTTP header
+	// +kubebuilder:validation:MaxItems=5
```

Comment:
can we enhance the doc string here to highlight that by default `By` and `Hash` and always included if this header is set ?

Reply:
this is done